app/uk/gov/hmrc/gform/auth/AuthService.scala

```diff
@@ -32,17 +35,19 @@ class AuthService {
       .fold("")(_.getIdentifier(EEITTAuthConfig.nonAgentIdName).fold("")(_.value))
   }
 
-  def evaluateSubmissionReference(expression: TextExpression, retrievals: Retrievals, data: Map[FormComponentId, Seq[String]]): String = {
+  def evaluateSubmissionReference(expression: TextExpression, retrievals: Retrievals, formTemplate: FormTemplate,
+    data: Map[FormComponentId, Seq[String]])(implicit hc: HeaderCarrier): Future[String] = {
```

Comment:
So this method returns `customerId`, but it is really hard to see. Once we would introduced `CustomerId` value class, return method `Future[CustomerId]` would be more pleasant to read and understand.